misc-scripts/xref_mapping/XrefParser/RefSeqCoordinateParser.pm

```diff
@@ -23,405 +23,302 @@ use strict;
-      if (defined $otherf_dba) { $otherf_dba->dnadb($core_dba); }
+  $species_name //= shift @{$id2name{$species_id}};
+
+  # prepare registry and core/otherfeatures dba
+  my $registry = "Bio::EnsEMBL::Registry";
+  my ($core_dba, $otherf_dba);
+
+  # for ensembl project, use provided connection details or default to staging
+  if ( $file_params->{project} eq ('ensembl') ) {
+    if (!defined $file_params->{host}) {
+      $file_params->{host} = 'mysql-ens-sta-1';
+      $file_params->{port} = '4519';
+      $file_params->{user} = 'ensro';
     }
-      
-
-  } elsif (defined $project && $project eq 'ensemblgenomes') {
-      $registry->load_registry_from_multiple_dbs( 
-	  {
-	      -host     => 'mysql-eg-staging-1.ebi.ac.uk',
-	      -port     => 4160,
-	      -user     => 'ensro',
-	  },
-	  {
-	      -host     => 'mysql-eg-staging-2.ebi.ac.uk',
-	      -port     => 4275,
-	      -user     => 'ensro',
-	  },
- 
-      );
-      $core_dba = $registry->get_DBAdaptor($species_name,'core');
-      $otherf_dba = $registry->get_DBAdaptor($species_name, 'otherfeatures');     
-
-  } elsif (defined $db) {
+    $registry->load_registry_from_db(
+      '-host' => $file_params->{host},
+      '-port' => $file_params->{port},
+      '-user' => $file_params->{user},
+    );
+    $core_dba = $registry->get_DBAdaptor($species_name, 'core');
+    if (!defined $file_params->{ofhost}) {
+      $file_params->{ofhost} = 'mysql-ens-sta-1';
+      $file_params->{ofport} = '4519';
+      $file_params->{ofuser} = 'ensro';
+    }
+    $registry->load_registry_from_db(
+      '-host' => $file_params->{ofhost},
+      '-port' => $file_params->{ofport},
+      '-user' => $file_params->{ofuser},
+    );
+    $otherf_dba = $registry->get_DBAdaptor($species_name, 'otherfeatures');
+    $otherf_dba->dnadb($core_dba);
+  # for ensemblgenomes project, use staging and ignore any connection details provided
+  } elsif ( $file_params->{project} eq ('ensemblgenomes') ) {
+    $registry->load_registry_from_multiple_dbs( {
+      '-host' => 'mysql-eg-staging-1.ebi.ac.uk',
+      '-port' => '4160',
+      '-user' => 'ensro',
+    }, {
+      '-host' => 'mysql-eg-staging-2.ebi.ac.uk',
+      '-port' => '4275',
+      '-user' => 'ensro',
+    } );
+    $core_dba = $registry->get_DBAdaptor($species_name, 'core');
+    $otherf_dba = $registry->get_DBAdaptor($species_name, 'otherfeatures');
+  # if no project but db provided, use that
+  } else {
     $otherf_dba = $db;
     $core_dba = $db->dnadb();
-  } else {
-      die("Missing or unsupported project value. Supported values: ensembl, ensemblgenomes");
   }
 
-## Not all species have an otherfeatures database, skip if not found
+  # Not all species have an otherfeatures database, error if not found
   if (!$otherf_dba) {
-    print STDERR "No otherfeatures database for $species_name, skipping import for refseq_import data\n";
-    return;
+    croak "No otherfeatures database found for $species_name\n";
```

Comment:
Similar comment here as for a missing refseq_import analysis
Some species will not have an otherfeatures database, in which case this parser should not be run but the rest of the pipeline should run without failing.
In the eHive pipeline, the parsing job should not be created if no otherfeatures database is available, so hopefully we would not hit this case. If we want to maintain backwards compatibility with the xref_parser.pl script though, we need to allow the parser to exit without error if no otherfeatures database is found.